k8s/k8s.go

```diff
@@ -2062,6 +2080,26 @@ func (k *k8sOps) IsPodBeingManaged(pod v1.Pod) bool {
 	return false
 }
 
+func (k *k8sOps) ValidateDeletedPod(podName string, podNS string) error {
+	if err := k.initK8sClient(); err != nil {
+		return err
+	}
+
+	if podName == "" {
+		return fmt.Errorf("cannot validate pod without service pod")
+	}
+
+	_, err := k.client.CoreV1().Pods(podNS).Get(podName, meta_v1.GetOptions{})
+	if err != nil {
+		if matched, _ := regexp.MatchString(".+ not found", err.Error()); matched {
```

Comment:
instead of doing match string, check for the kubernetes error
"k8s.io/apimachinery/pkg/api/errors"
errors.IsNotFound(err)

Reply:
Fixed.